src/lib/metrics.js

```diff
@@ -0,0 +1,99 @@
+// Track recent tabs woken to see whether they're later closed or focused
+let unseenWakeHistory = {};
+let seenWakeHistory = {};
+
+// BroadcastChannel for sending metrics pings to Test Pilot add-on
+let pingChannel = null;
+
+// Name of the metrics BroadcastChannel expected by the Test Pilot add-on
+const TESTPILOT_TELEMETRY_CHANNEL = 'testpilot-telemetry';
```

Comment:
What I've got here now is just basically copying what the Tracking Protection experiment does now. But for Telemetry vs GA (or both), these are all things that will hopefully be addressed with [the pending metrics broker solution from Test Pilot](https://github.com/mozilla/testpilot/issues/1735). Not sure what that will look like yet, though